src/ui/widgets/Tabs/tabContainer.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import log from "loglevel";
+
+import { Widget } from "../widget";
+import { WidgetPropType } from "../widgetProps";
+import { registerWidget } from "../register";
+import {
+  InferWidgetProps,
+  StringOrNumPropOpt,
+  BorderPropOpt,
+  ColorPropOpt
+} from "../propTypes";
+import { errorWidget } from "../EmbeddedDisplay/embeddedDisplay";
+import { parseJson } from "../EmbeddedDisplay/jsonParser";
+import { widgetDescriptionToComponent } from "../createComponent";
+
+import classes from "./navigationTabs.module.css";
+
+export const TabContainerProps = {
+  tabs: PropTypes.objectOf(PropTypes.object).isRequired,
+  maxHeight: StringOrNumPropOpt,
+  maxWidth: StringOrNumPropOpt,
+  minHeight: StringOrNumPropOpt,
+  border: BorderPropOpt,
+  backgroundColor: ColorPropOpt
+};
+
+export const TabContainerComponent = (
+  props: InferWidgetProps<typeof TabContainerProps>
+): JSX.Element => {
+  const [childIndex, setIndex] = useState(0);
+
+  console.log(props.tabs);
+
+  try {
+    // TODO: Find out if this repeated calculation can be done in the useMemo hook for measurable performance gains
+    const children = Object.values(props.tabs).map((child, index) =>
+      widgetDescriptionToComponent(
+        parseJson(JSON.stringify(child), "pva"),
+        index
+      )
```

Comment:
@willrogers I don't feel great about stringifying this object just to parse it again, but I think it is acceptable at the moment. If we start finding more places where this pops up then perhaps we should look for another way to do it.

This method keeps the JSON parsing clean (other solutions I thought of involved changing the JSON parser to check for the special case of tabs).

Let me know if you think there is a better way to do this.